### PR TITLE
Task 6 - class Game

### PR DESCRIPTION
**Pull request by dawidjozefczak**

### CODE REVIEW

File: shm/src/game.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            std::cout << "You do not have enough space on your ship \n";
+        }
+        store_.printCargo();
+    }
+}
+
+void Game::PrintCargo() {
+    std::"You have:\n";
+    player_->printShipCargo();
+}
```
Comment (mariuszlisowski) [inline shm/src/game.cpp]: newline

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]
Good job Dawid!

**Review by mariuszlisowski — APPROVED** [review]

File: shm/src/game.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        Cargo* cargo = player_.getCargo(--product);
+        Store::Response response = store_.sell(cargo, amount, player_.get());
+        switch (response) {
+        case Store::Response::done:
+            std::cout << "Sell " << amount << " " << product << "\n";
+            return;
+        case Store::Response::lack_of_cargo:
+            std::cout << "There is no enough cargo to sell \n";
+        case Store::Response::lack_of_space:
+            std::cout << "There is no enough space in storess \n";
```
Comment (m-kop) [inline shm/src/game.cpp:129]: no return no break here?

File: shm/src/game.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        switch (response) {
+        case Store::Response::done:
+            std::cout << "Buy " << amount << " " << product << "\n";
+            return;
+        case Store::Response::lack_of_cargo:
+            std::cout << "There is no enough cargo to buy \n";
+        case Store::Response::lack_of_money:
+            std::cout << "You do not have enough money \n";
+        case Store::Response::lack_of_space:
+            std::cout << "You do not have enough space on your ship \n";
```
Comment (m-kop) [inline shm/src/game.cpp:152]: ditto

File: shm/src/game.cpp [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            break;
+        }
+        std::cout << "\nWrong desination!\n";
+    }
+}
+
+void Game::Sell() {
+    while (true) {
+        player_->printShipCargo();
+        std::cout << store_ << "Select product and quantity: ";
```
Comment (m-kop) [inline shm/src/game.cpp:117]: store_ is a unique_ptr and << operator is not handling that

**Review by m-kop — CHANGES_REQUESTED** [review]

**Pull request closed without merging**